Automattic/wp-calypso
# composite-checkout: Convert to TypeScript

#### Changes proposed in this Pull Request

This converts the `@automattic/composite-checkout` package to TypeScript. It should not cause any user-facing changes, but it does restructure some of the code.

The only files not converted by this PR are the demo and the payment methods, the latter of which are quite complex and I think would be better served by having a PR dedicated to them alone.

This also does not explicitly export any of the types we've added here apart from the declaration files that are created. I don't know what the normal way is to do this. I see that some packages in the monorepo include the `dist` file in the committed code, but I don't fully understand the implications of this or the best practices, so I felt that doing anything like this should be decided later.

Fixes https://github.com/Automattic/wp-calypso/issues/45000 (for the most part).

Fixes https://github.com/Automattic/wp-calypso/issues/45001

#### Testing instructions

This touches many aspects of checkout, so it's worth running through some of the common testing scenarios, although for the most part the unit and e2e tests should serve to prevent regressions.

Since this also touches the stripe library bundled in the package, and the only place that library is currently used (outside the demo) is in the 3DS handling of the `createExistingCardMethod` function, we should notably test the existing card payment method to make sure it works with 3DS cards.
